metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java

```diff
@@ -97,95 +112,91 @@ public void set(String depositUUID, DepositField field, String value) {
 	 * @return true if lock acquired
 	 */
 	public boolean addSupervisorLock(String depositUUID, String owner) {
-		Jedis jedis = getJedisPool().getResource();
-		Long result = jedis.hsetnx(DEPOSIT_STATUS_PREFIX+depositUUID, DepositField.lock.name(), owner);
-		getJedisPool().returnResource(jedis);
-		return result == 1;
+		try (Jedis jedis = getJedisPool().getResource()) {
+			Long result = jedis.hsetnx(DEPOSIT_STATUS_PREFIX + depositUUID,
+					DepositField.lock.name(), owner);
+			return result == 1;
+		}
 	}
 
 	/**
 	 * Unlocks the given deposit, allowing a new supervisor to manage it.
 	 * @param depositUUID
 	 */
 	public void removeSupervisorLock(String depositUUID) {
-		Jedis jedis = getJedisPool().getResource();
-		jedis.hdel(DEPOSIT_STATUS_PREFIX+depositUUID, DepositField.lock.name());
-		getJedisPool().returnResource(jedis);
+		try (Jedis jedis = getJedisPool().getResource()) {
+			jedis.hdel(DEPOSIT_STATUS_PREFIX + depositUUID, DepositField.lock.name());
+		}
 	}
 
 	public DepositState getState(String depositUUID) {
 		DepositState result = null;
-		Jedis jedis = getJedisPool().getResource();
-		String state = jedis.hget(DEPOSIT_STATUS_PREFIX+depositUUID, DepositField.state.name());
-		try {
+		
+		try (Jedis jedis = getJedisPool().getResource()) {
+			String state = jedis.hget(DEPOSIT_STATUS_PREFIX + depositUUID, DepositField.state.name());
 			result = DepositState.valueOf(state);
-		} catch(NullPointerException e) {
-		} catch(IllegalArgumentException e) {
+		} catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
I know it's how the code was before your changes, but do we really want to be catching a NPE here? Seems like testing for null and then having a path where the code takes appropriate action might be better.